src/Core/ShellUtility.re

```diff
@@ -11,19 +13,38 @@ module Internal = {
       "";
     };
 
+  let runCommand = (cmd) => {
+      Log.infof(m => m("Running cmd: %s", cmd));
+      let (stdOut, stdIn, stdErr) =
+        Unix.open_process_full(cmd, [||]);
+      let ret = input_line(stdOut);
+      Log.infof(m => m("Received output: %s", ret));
+      let () = close_in(stdOut);
+      let () = close_out(stdIn);
+      let () = close_in(stdErr);
+      ret;
+  };
+
+  let discoverLinuxShell = () =>  {
+    try({
+      let user = Sys.getenv("USER");
+      let userShell = runCommand("getent passwd " ++ user ++ " | awk -F: '{print $NF}'");
+      userShell
```

Comment:
Why not just use [`Unix.getpwuid`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Unix.html#VALgetpwuid)?

Reply:
Oh, nice find - I didn't know about those APIs. Thanks for the suggestion 👍

Reply:
Np :) I mentioned it in our meeting last monday though ;)